packages/eslint-plugin/package.json

```diff
@@ -98,12 +98,8 @@
   },
   "peerDependencies": {
     "@typescript-eslint/parser": "^8.0.0 || ^8.0.0-alpha.0",
-    "eslint": "^8.57.0 || ^9.0.0"
-  },
-  "peerDependenciesMeta": {
-    "typescript": {
-      "optional": true
-    }
```

Comment:
Please leave this in.
We do not want to warn about missing peer dependencies in cases where our plugin is installed as part of a shared config but not used.

Reply:
You have also add other `peerDependenciesMeta` dependencies. Beyond `typescript`, what are optional dependencies for shared config?

Reply:
For context we intentionally had an optional peer dep just for TS here because it meant that configs could in turn have an optional dependency on our tooling without creating log spam for end users that didn't use TS. However with flat configs this isn't possible any more - which means that this is now not really required.

HOWEVER. Removing this could cause log spam for these existing eslintrc usecases. As such I think we should leave it in. Once we drop eslintrc support (and I suppose ESLint v8 support) then we can also remove the optional.

---

Full explaination:

In an eslintrc file you never directly imported a plugin or parser -- it was done *_lazily_* by ESLint.
This meant that if you used a plugin or parser in an `overrides` block and the block didn't match the files you were linting then ESLint wouldn't ever load the plugin/parser.

However in flat configs you import the plugin/parser directly in the config file. This means that no matter what you do you are always importing the entire dependency graph of every single parser and plugin you use.

This is obviously going to be wasteful AND slower.

But it also means that a config can no longer have a "conditional" block that only depends on something if it's used.

EG if a shared eslintrc config defined like
```
module.exports = {
  overrides: [
    {
      files: ["*.ts"],
      extends: ["plugin:@typescript-eslint/recommended"],
    },
  ],
};
```
Then our plugin and parser would only load if TS files exist. This meant that our packages were optional dependencies of the config. Additionally (and more importantly) it meant that our peer dep on `typescript` was also optional as it is only required if TS files exist.

OTOH if a flat shared config defines like
```
export default [
  {
    files: ["**/*.ts],
    extends: [tseslint.configs.recommended],
  },
];
``` 
Then our plugin and parser are eagerly loaded and thus they are hard dependencies of the config. Additionally our peer dep on `typescript` is no longer optional because our tooling `require('typescript')` at the top level and thus it's now a required peer dep.

All of this really means that declaring an **_optional_** peer dependency on TS is no longer correct because it's not optional!

Reply:
This is the deps tree:
```console
typescript-eslint
└─ @typescript-eslint/utils
|  └─ @typescript-eslint/typescript-estree
|     └─ ts-api-utils@1.3.0 (peer dep typescript >=4.2.0)
└─ @typescript-eslint/parser
   └─ @typescript-eslint/typescript-estree
      └─ ts-api-utils@1.3.0 (peer dep typescript >=4.2.0)
```
`ts-api-utils` uses `typescript` under the hood.
So everywhere you use `ts-api-utils` dependency, you need to install `typescript` also (as dependency or peer dependency).

Expected result:
```console
typescript-eslint (peer dep typescript >=4.2.0)
└─ @typescript-eslint/utils (peer dep typescript >=4.2.0)
|  └─ @typescript-eslint/typescript-estree (peer dep typescript >=4.2.0)
|     └─ ts-api-utils@1.3.0 (peer dep typescript >=4.2.0)
└─ @typescript-eslint/parser (peer dep typescript >=4.2.0)
   └─ @typescript-eslint/typescript-estree (peer dep typescript >=4.2.0)
      └─ ts-api-utils@1.3.0 (peer dep typescript >=4.2.0)
```